File: HexChat/logpm.py
import xchat as hexchat

__module_name__ = "LogPMs"
__module_author__ = "TingPing"
__module_version__ = "2"
__module_description__ = "Auto log pm's"

def open_cb(word, word_eol, userdata):
	chan = hexchat.get_info('channel')
	# Assume nick if not prefixed with #
	# Ignore ZNC and Python
	# Use existing pref for nicks I usually ignore (i.e. chanserv)
	if chan and chan[0] != '#' \
		and chan[0] != '*' \
		and chan != '>>python<<' \
		and chan not in hexchat.get_prefs('irc_no_hilight').split(','):
		hexchat.command('chanopt -quiet text_logging on')
		hexchat.command('chanopt -quiet text_scrollback on')

hexchat.hook_print("Open Context", open_cb)

File: HexChat/logpm.lua
-- SPDX-License-Identifier: MIT
hexchat.register('LogPM', '2', 'Automatically log private queries')

hexchat.hook_print('Open Context', function (args)
	-- We only want queries
	if hexchat.props['type'] ~= 3 then
		return
	end

	-- Ignore empty tabs, znc queries, and scripting consoles
	if not hexchat.get_info('channel'):match('^[%*%(>]') then
		hexchat.command('chanopt -quiet text_logging on')
	end
end)
